registry/remote/auth/scope.go

```diff
@@ -93,7 +119,62 @@ func AppendScopes(ctx context.Context, scopes ...string) context.Context {
 // GetScopes returns the scopes in the context.
 func GetScopes(ctx context.Context) []string {
```

Comment:
Should we deprecate the old `GetScopes`, WithScopes` and `AppendScopes`? They are not safe enough but they a little bit convenient than the new ones in scenarios where there is only one host or two hosts share the same scope.

Reply:
Nope. We can consider `GetScopes()` as a global version of `GetScopesForHost()`.